### PR TITLE
`sage.rings.number_field`: Modularization fixes, doctest cosmetics, `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
- Avoiding going through symbolic square roots in `.S_unit_solver`
- `# needs sage.symbolic`, `# needs sage.libs.*`, `# needs sage.rings.finite_rings` etc.
- Reformatting doctests to fit on the screen
- Codestyle fixes in doctests 
- Block tags for `# optional - magma`
- Handle import errors in `NumberField._pushout_`

This is
- Part of: #29705 
- Cherry-picked from: #35095 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/number_field/number_field.py [hunk 1]
```diff
@@ -1547,16 +1549,17 @@ def _magma_polynomial_(self, magma):
...
+            sage: # optional - magma
+            sage: R.<x> = QQ[]
+            sage: K.<a> = NumberField(x^3 + 2)
+            sage: K._magma_polynomial_(magma)
             x^3 + 2
-            sage: magma2=Magma()                                                 # optional - magma
-            sage: K._magma_polynomial_(magma2)                                   # optional - magma
+            sage: magma2=Magma()
+            sage: K._magma_polynomial_(magma2)
             x^3 + 2
```
Comment (kwankyu) [inline src/sage/rings/number_field/number_field.py:1559]: Spaces around `=`

File: src/sage/rings/number_field/number_field.py [hunk 2]
```diff
@@ -1880,17 +1887,17 @@ def _convert_non_number_field_element(self, x):
         will convert to the number field, e.g., this one in
         characteristic 7::
 
-            sage: f = GF(7)['y']([1,2,3]); f                                            # optional - sage.rings.finite_rings
+            sage: f = GF(7)['y']([1,2,3]); f                                            # needs sage.rings.finite_rings
             3*y^2 + 2*y + 1
```
Comment (kwankyu) [inline src/sage/rings/number_field/number_field.py:1891]: This is not needed anymore for prime finite fields?

Comment (mkoeppe) [reply src/sage/rings/number_field/number_field.py:1891]: yes, for small enough prime finite fields

**Review by kwankyu — APPROVED** [review]
LGTM.

File: src/sage/rings/number_field/number_field.py [hunk 3]
```diff
@@ -7343,7 +7368,7 @@ def S_unit_group(self, proof=None, S=None):
 
         INPUT:
 
-        - ``proof`` (bool, default True) flag passed to ``pari``.
+        - ``proof`` (bool, default True) flag passed to PARI.
```
Comment (dcoudert) [inline src/sage/rings/number_field/number_field.py]:
```diff
-        - ``proof``  (bool, default True) flag passed to PARI.
+        - ``proof`` -- bool (default: ``True``); flag passed to PARI
```

File: src/sage/rings/number_field/number_field.py [hunk 4]
```diff
@@ -11273,33 +11311,35 @@ def _log_gen(self, x):
...
              0.998210129767736 + 0.0598041539450313*I)
-            sage: K._log_gen(zeta105^3)                                                 # optional - sage.symbolic
+            sage: K._log_gen(zeta105^3)
             81
-            sage: zeta105^3, CC(a)^81                                                   # optional - sage.symbolic
+            sage: zeta105^3, CC(a)^81
             (0.983929588598630 + 0.178556894798637*I,
              0.983929588598631 + 0.178556894798635*I)
 
             sage: K.<a> = CyclotomicField(5, embedding=None)
```
Comment (dcoudert) [inline src/sage/rings/number_field/number_field.py:11335]: shouldn't we add `# needs sage.symbolic` here ? it's not the same bloc due to blank line

File: src/sage/rings/number_field/number_field_element.pyx [hunk 5]
```diff
@@ -582,19 +582,19 @@ cdef class NumberFieldElement(NumberFieldElement_base):
         EXAMPLES::
 
             sage: F = CyclotomicField(8)
-            sage: F.gen()._libgap_()                                                    # optional - sage.libs.gap
+            sage: F.gen()._libgap_()                                                    # needs sage.libs.gap
```
Comment (dcoudert) [inline src/sage/rings/number_field/number_field_element.pyx:585]: block scope

**Review by dcoudert — CHANGES_REQUESTED** [review]

File: src/sage/rings/number_field/number_field.py (hunk 3 above)
Comment (mkoeppe) [reply src/sage/rings/number_field/number_field.py]: Done in 69d6265100

File: src/sage/rings/number_field/number_field.py (hunk 4 above)
Comment (mkoeppe) [reply src/sage/rings/number_field/number_field.py:11335]: Above needed `sage.symbolic` because of `exp`, `pi`, `i`. This one here doesn't

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks both!

**Pull request merged**